File: lib/common/dto/coordinate/coordinate_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:line_flux/common/model/coordinate/coordinate.dart';

part 'coordinate_dto.freezed.dart';
part 'coordinate_dto.g.dart';

//todo
@freezed
class CoordinateDTO with _$CoordinateDTO {

  const factory CoordinateDTO({
    required double latitude,
    required double longitude,
    required double altitude,
  }) = _CoordinateDTO;

  const CoordinateDTO._();

  factory CoordinateDTO.fromJson(Map<String, dynamic> json) => _$CoordinateDTOFromJson(json);

  factory CoordinateDTO.fromDataModel(Coordinate model) {
    return CoordinateDTO(latitude: model.latitude, longitude: model.longitude, altitude: model.altitude);
  }

  Coordinate toDataModel() => Coordinate(latitude: latitude, longitude: longitude, altitude: altitude);
}

File: android/app/src/main/kotlin/com/line_flux/line_flux/dto/coordinate/CoordinateDTO.kt
package com.line_flux.line_flux.dto.coordinate

import com.kotlin.sensor_drawing_plugin.coordinate.Coordinate

data class CoordinateDTO(
  var latitude: Double = 0.0,
  var longitude: Double = 0.0,
  var altitude: Double = 0.0,
) {
  companion object {
    fun fromDataModel(coordinate: Coordinate): CoordinateDTO {
      return _root_ide_package_.com.line_flux.line_flux.dtos.coordinate.CoordinateDTO(
        latitude = coordinate.latitude,
        longitude = coordinate.longitude,
        altitude = coordinate.altitude
      )
    }
  }

  fun toDataModel(): Coordinate {
    return Coordinate(latitude, longitude, altitude)
  }
}
